### PR TITLE
feat: add guid to each objects metadata on export

### PR DESCRIPTION
**Pull request by donutboyy**
needed by https://github.com/area28technologies/io.area28.unity/pull/44

### CODE REVIEW

File: Runtime/Scripts/Schema/JsonWriter.cs [hunk 1]
```diff
@@ -49,6 +49,15 @@ public void AddProperty(string name)
             m_Separation = false;
         }
 
+		public void AddGuid(string guid)
+        {
+            Separate();
+            m_Stream.Write("\"extras\": {\"guid\": \"");
+            m_Stream.Write(guid);
+            m_Stream.Write("\"}");
```
Comment (jackylauu) [inline Runtime/Scripts/Schema/JsonWriter.cs]:
```suggestion
            m_Stream.Write($@"""extras"": {{""guid"": ""{guid}""}}");
```

**Review by vayan — APPROVED** [review]
seems good!

**Comment by donutboyy** [comment]
There seems to be a cleaner way to do this that I missed 😅, tried it and it works well! Also includes some settings for the export
https://github.com/atteneder/glTFast/blob/main/Documentation~/ExportRuntime.md

File: Runtime/Scripts/Export/GameObjectExport.cs [hunk 2]
```diff
@@ -172,12 +172,15 @@ bool AddGameObject(GameObject gameObject, List<Material> tempMaterials, out int
 
             if (onIncludedLayer || children != null)
             {
+                gameObject.TryGetComponent<A28.GuidComponent>(out A28.GuidComponent guidComponent);
+                string guid = guidComponent != null ? guidComponent.GuidString : null;
```
Comment (jackylauu) [inline Runtime/Scripts/Export/GameObjectExport.cs]:
```suggestion
                string guid = guidComponent?.GuidString;
```

File: Runtime/Scripts/Export/GltfWriter.cs [hunk 3]
```diff
@@ -1891,13 +1893,15 @@ NativeArray<byte> outputStream
...
+            string name = null,
+            string guid = null
             )
         {
             var node = new Node
             {
-                name = name,
+                name = name
             };
+			node.guid = guid;
```
Comment (jackylauu) [inline Runtime/Scripts/Export/GltfWriter.cs]:
```suggestion
                name = name,
                guid = guid,
            };
```

File: Runtime/Scripts/Schema/JsonWriter.cs [hunk 4]
```diff
@@ -49,6 +49,13 @@ public void AddProperty(string name)
             m_Separation = false;
         }
 
+		public void AddGuid(string guid)
```
Comment (jackylauu) [inline Runtime/Scripts/Schema/JsonWriter.cs]:
```suggestion
        public void AddGuid(string guid)
```

Comment (donutboyy) [reply Runtime/Scripts/Schema/JsonWriter.cs]: Maybe we should set up format checks

**Review by jackylauu — APPROVED** [review]
🙏

**Pull request merged**